python/ray/data/read_api.py

```diff
@@ -418,28 +419,15 @@ def read_images(
         >>> path = "s3://air-example-data-2/movie-image-small-filesize-1GB"
         >>> ds = ray.data.read_images(path)  # doctest: +SKIP
         >>> ds  # doctest: +SKIP
-        Dataset(num_blocks=200, num_rows=41979, schema={__value__: ArrowTensorType(shape=(386, 256, 3), dtype=uint8)})
-
-        If your images are arranged like:
-
-        .. code::
-
-            root/dog/xxx.png
-            root/dog/xxy.png
+        Dataset(num_blocks=200, num_rows=41979, schema={image: ArrowVariableShapedTensorType(dtype=uint8, ndim=3)})
 
-            root/cat/123.png
-            root/cat/nsdf3.png
+        If you also need the image file paths, set ``include_paths=True``.
 
-        Then you can include the labels by specifying a
-        :class:`~ray.data.datasource.partitioning.Partitioning`.
-
-        >>> import ray
-        >>> from ray.data.datasource.partitioning import Partitioning
-        >>> root = "example://tiny-imagenet-200/train"
-        >>> partitioning = Partitioning("dir", field_names=["class"], base_dir=root)
-        >>> ds = ray.data.read_images(root, size=(224, 224), partitioning=partitioning)  # doctest: +SKIP
+        >>> ds = ray.data.read_images(path, include_paths=True)  # doctest: +SKIP
         >>> ds  # doctest: +SKIP
-        Dataset(num_blocks=176, num_rows=94946, schema={image: TensorDtype(shape=(224, 224, 3), dtype=uint8), class: object})
```

Comment:
Hmm can we keep the partitioning example? We'd like to guide users towards this if it fits with their use case, rather than the `include_paths` route (which should be a last-resort escape hatch).